corehq/apps/user_importer/tasks.py

```diff
@@ -32,6 +34,9 @@ def _update_progress(value, start=0):
         'errors': group_results['errors'] + user_results['errors'],
         'rows': user_results['rows']
     }
+    upload_record = UserUploadRecord.objects.get(task_id=import_users_and_groups.request.id)
+    upload_record.status = results
+    upload_record.save()
```

Comment:
what if the upload fails midway?
Do we want to put this in a try except block and store whatever progress was made?

Reply:
The function itself `create_or_update_users_and_groups` has a lot of error handling that feels sufficient to me. I think the only thing it would catch is if the celery task itself died and wrapping it in a try would not catch that either. Do you know how ICDS ones fail midway? I think they are the main people that have issues with that

Reply:
The one that i looked at recently felt to get broken because of a deploy. That was the only thing that explained due to the error occurrence time matching the time of deploy and no record of any error in sentry to refer to.

Here, since the process is not transactional, the record would definitely miss out information about some users that will get added if the process fails midway. I am thinking of hard exceptions that are not accounted for and can happen. I understand if the process gets killed, there is nothing that can be done anyway

Reply:
I am not totally sure I understand. Where would the exception happen that this code would miss? If there is an exception in `create_or_update_users_and_groups` that will be caught, since everything is wrapped in a try, except block (both at the row level and one for the entire import). So if that function has an exception, the partial progress would be stored. The only times I can see that failing is if the process itself was terminated, likely what happened on that deploy, but that would fail regardless of whether we had another try in this function.

The only way to get partial progress in those cases would be to save this model each time a row is processed, but that seems horribly slow, especially for large imports.